### ISSUE TITLE
Transitive dependencies of path dependencies are not updated with `poetry update X`

### ISSUE THREAD

**Issue by md384** [python-poetry/poetry#5111]
- [ ] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [ ] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [ ] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: MacOS 11.2
- **Poetry version**: 1.1.12
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: [Gist](https://gist.github.com/md384/15f4be1ff0ed383f5f3a96dc82dd54e5)

## Issue

In my monorepo project, we have chains of path dependencies. For example, `lib_c` depends on `lib_b` which depends on `lib_a` - see [Gist](https://gist.github.com/md384/15f4be1ff0ed383f5f3a96dc82dd54e5) - which are defined in a single directory like

```
libs/
    lib_a/
        pyproject.toml
        ...
    lib_b/
        pyproject.toml
        ...
    lib_c/
        pyproject.toml
        ...
```

We have noticed that transitive dependencies of path dependencies are not updated when running `poetry update some_path_dependency`.

Taking the example `lib_c` depends on `lib_b` which depends on `lib_a`, if a dependency is added to `lib_a` (for example `toml` package), running `poetry update lib_a` in `lib_a` directory will update the `lib_b` `poetry.lock` adding `toml` as a dependency of `lib_a` like below.

```
[[package]]
name = "lib-a"
version = "0.1.0"
description = ""
category = "main"
optional = false
python-versions = "^2.7"
develop = true

[package.dependencies]
toml = "^0.10.2"

[package.source]
type = "directory"
url = "../lib_a"

[[package]]
name = "toml"
version = "0.10.2"
description = "Python Library for Tom's Obvious, Minimal Language"
category = "main"
optional = false
python-versions = ">=2.6, !=3.0.*, !=3.1.*, !=3.2.*"
```

However when running `poetry update lib_b` in `lib_c`, `toml` is __not__ installed or referenced in `lib_c`s lockfile. See below.

```
[[package]]
name = "lib-a"
version = "0.1.0"
description = ""
category = "main"
optional = false
python-versions = "^2.7"
develop = true

[package.source]
type = "directory"
url = "../lib_a"

[[package]]
name = "lib-b"
version = "0.1.0"
description = ""
category = "main"
optional = false
python-versions = "^2.7"
develop = true

[package.dependencies]
lib_a = {path = "../lib_a", develop = true}

[package.source]
type = "directory"
url = "../lib_b"
```

This results in `ImportError`s when trying to use `lib_c`.

We have found a workaround which is to run `poetry update lib_a lib_b` on `lib_c` but this seems like a bug. We would expect that `poetry update some_package` would capture and update all nested dependencies in the lockfile too.

Running `poetry update` on `lib_c` (i.e. without specifying the packages to update) also works however no desirable for us since we want to update our path dependencies only.

We also tried `poetry lock --no-update` on `lib_c` but that does not add `toml` to the `lib_c` lockfile - that is likely expected.

**Comment by netoront** [comment]
@md384 your last workaround attempt demonstrates the issue I'm having in my monorepo: `poetry lock --no-update` doesn't seem to follow path dependencies.

TL;DR: Bumping the patch version of the package whose dependencies changed might be a workaround for your problem.

In my testing, `poetry lock --no-update` followed path dependencies *only if the path dependencies' versions were different from the versions in `poetry.lock`*. It looks like Poetry assumes "version hasn't changed" implies "dependencies haven't changed". This makes a lot of sense for non-path dependencies, and *could* make sense for path dependencies. But when developing in a monorepo, changing and installing dependencies is done earlier than bumping package versions, and the latter is often done in CI, not in the dev's local clone.

My kludge: I've written a script that runs before `poetry lock --no-update` that replaces the version strings of the monorepo packages with `"0"` in `poetry.lock`.
